[cec] Add settings for configuring button repeats

popcornmix opened the pull request:
<!--- Provide a general summary of your change in the Title above -->

## Description
Add settings for configuring CEC button repeats

## Motivation and Context
CEC remotes repeat at the rate the TV vendor chooses. This may be just a couple of times a second.
This is slow compared to other kodi controllers (lirc, keyboard, game controller) and makes kodi seem laggy.

I have added some changes that are now present in libCEC 4 that allow the CEC repeat messages to be ignored and replaced with messages more like a keyboard generates. The controls are:
* Remote button press delay before repeating (ms)
* Remote button press repeat rate (ms)
* Remote button press release time (ms)

CEC is a poorly implemented standard, and behaviour between different vendors is not consistent which makes these settings necessary. A CEC source may:

Send pressed on press and released on release.
Send pressed on press and periodically and released on release.
Send pressed on press and periodically and nothing on release.

The "release" setting can allow a timeout to be set to assume button is released after, for sources in the third category. This is not ideal as it means the release lags by the amount set. If the periodic repeat rate is high, then the release time can be set lower to reduce the lag.

If the repeat delay is left at 0 (the default) then the behaviour is unchanged.
Typically just changing the repeat delay to, say 150ms will enable this feature and scrolling becomes much quicker. The other options only need adjusting in a minority of cases where source doesn't behave as desired.

## How Has This Been Tested?
Has been present in Raspberry Pi builds for over a year.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

popcornmix commented:
Ping @opdenkamp

wrxtasy commented:
I've actually back ported this PR into AMLogic LibreELEC 8.0 Krypton that has LibCEC 4 support and results are noticeable improvements in CEC remote response times. 👍

Rechi commented:
@popcornmix this needs a rebase

popcornmix committed:
[cec] Add settings for configuring button repeats

popcornmix committed:
[cec] Don't discard buttons when repeat mode is enabled

popcornmix committed:
[cec] Drop CEC_DOUBLE_TAP_TIMEOUT_MS_OLD code

Kodi won't even build with libcec 3, so supporting a libcec 2.2 setting is of no value

Rechi commented:
@popcornmix @opdenkamp ping
